### PR TITLE
Replace internal/kubeapi functions with direct client calls

### PR DESCRIPTION
**Pull request by cbandy**
This is one step towards using the interfaces provided by k8s.io/client-go that can facilitate unit/local tests.

**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change, as applicable?
 - [x] Have you tested your changes on all related environments with successful results, as applicable?

**Type of Changes:**

 
 - [x] Bug fix (non-breaking change which fixes an issue)

**Other Information**:
- Wait for these to merge first:
   - [x] #1649

### CODE REVIEW

File: internal/apiserver/pgoroleservice/pgoroleimpl.go [hunk 1]
```diff
@@ -296,7 +302,7 @@ func deleteRoleFromUsers(clientset *kubernetes.Clientset, roleName string) error
 			}
 
 			s.Data[pgouserservice.MAP_KEY_ROLES] = []byte(resultingRoleString)
-			err = kubeapi.UpdateSecret(clientset, &s, apiserver.PgoNamespace)
+			_, err = clientset.CoreV1().Secrets(apiserver.PgoNamespace).Update(&s)
 			if err != nil {
```
Comment (jkatz) [inline internal/apiserver/pgoroleservice/pgoroleimpl.go:306]:
```suggestion
			if _, err := clientset.CoreV1().Secrets(apiserver.PgoNamespace).Update(&s); err != nil {
```

File: internal/apiserver/pgouserservice/pgouserimpl.go [hunk 2]
```diff
@@ -157,10 +159,10 @@ func DeletePgouser(clientset *kubernetes.Clientset, deletedBy string, request *m
 		secretName := "pgouser-" + v
 		log.Debugf("DeletePgouser %s deleted by %s", secretName, deletedBy)
 
-		if _, err := kubeapi.GetSecret(clientset, secretName, apiserver.PgoNamespace); err != nil {
+		if _, err := clientset.CoreV1().Secrets(apiserver.PgoNamespace).Get(secretName, metav1.GetOptions{}); err != nil {
 			resp.Results = append(resp.Results, secretName+" not found")
 		} else {
-			err = kubeapi.DeleteSecret(clientset, secretName, apiserver.PgoNamespace)
+			err = clientset.CoreV1().Secrets(apiserver.PgoNamespace).Delete(secretName, &metav1.DeleteOptions{})
 			if err != nil {
```
Comment (jkatz) [inline internal/apiserver/pgouserservice/pgouserimpl.go:166]:
```suggestion
			if err := clientset.CoreV1().Secrets(apiserver.PgoNamespace).Delete(secretName, &metav1.DeleteOptions{}); err != nil {
```

File: internal/apiserver/pgouserservice/pgouserimpl.go [hunk 3]
```diff
@@ -240,7 +242,7 @@ func UpdatePgouser(clientset *kubernetes.Clientset, updatedBy string, request *m
 	}
 
 	log.Info("Updating secret for: ", request.PgouserName)
-	err = kubeapi.UpdateSecret(clientset, secret, apiserver.PgoNamespace)
+	_, err = clientset.CoreV1().Secrets(apiserver.PgoNamespace).Update(secret)
 	if err != nil {
```
Comment (jkatz) [inline internal/apiserver/pgouserservice/pgouserimpl.go:246]:
```suggestion
	if _, err := clientset.CoreV1().Secrets(apiserver.PgoNamespace).Update(secret); err != nil {
```

File: internal/apiserver/root.go [hunk 4]
```diff
@@ -479,7 +479,7 @@ func generateTLSCert(certPath, keyPath string) error {
 	newSecret.Data[corev1.TLSPrivateKeyKey] = privateKeyBytes
 	newSecret.Type = corev1.SecretTypeTLS
 
-	err = kubeapi.CreateSecret(Clientset, &newSecret, PgoNamespace)
+	_, err = Clientset.CoreV1().Secrets(PgoNamespace).Create(&newSecret)
 	if err != nil {
```
Comment (jkatz) [inline internal/apiserver/root.go:483]:
```suggestion
	if _, err := Clientset.CoreV1().Secrets(PgoNamespace).Create(&newSecret); err != nil {
```

File: internal/apiserver/scheduleservice/scheduleimpl.go [hunk 5]
```diff
@@ -179,8 +179,8 @@ func CreateSchedule(request *msgs.CreateScheduleRequest, ns string) msgs.CreateS
 		}
 
 		log.Debug("Getting configmap..")
-		_, exists := kubeapi.GetConfigMap(apiserver.Clientset, schedule.Name, schedule.Namespace)
-		if exists {
+		_, err = apiserver.Clientset.CoreV1().ConfigMaps(schedule.Namespace).Get(schedule.Name, metav1.GetOptions{})
+		if err == nil {
```
Comment (jkatz) [inline internal/apiserver/scheduleservice/scheduleimpl.go:183]:
```suggestion
		if _, err := apiserver.Clientset.CoreV1().ConfigMaps(schedule.Namespace).Get(schedule.Name, metav1.GetOptions{}); err == nil {
```

File: internal/apiserver/scheduleservice/scheduleimpl.go [hunk 6]
```diff
@@ -194,15 +194,15 @@ func CreateSchedule(request *msgs.CreateScheduleRequest, ns string) msgs.CreateS
...
 				Name:   schedule.Name,
 				Labels: labels,
 			},
 			Data: data,
 		}
 
 		log.Debug("Creating configmap..")
-		err = kubeapi.CreateConfigMap(apiserver.Clientset, configmap, schedule.Namespace)
+		_, err = apiserver.Clientset.CoreV1().ConfigMaps(schedule.Namespace).Create(configmap)
 		if err != nil {
```
Comment (jkatz) [inline internal/apiserver/scheduleservice/scheduleimpl.go:206]:
```suggestion
		if _, err := apiserver.Clientset.CoreV1().ConfigMaps(schedule.Namespace).Create(configmap); err != nil {
```

File: internal/controller/job/rmdatahandler.go [hunk 7]
```diff
@@ -52,16 +52,20 @@ func (c *Controller) handleRMDataUpdate(job *apiv1.Job) error {
...
 	}
 
 	removed := false
 	for i := 0; i < deleteRMDataJobMaxTries; i++ {
 		log.Debugf("sleeping while job %s is removed cleanly", job.Name)
 		time.Sleep(time.Second * time.Duration(deleteRMDataJobDuration))
-		_, found := kubeapi.GetJob(c.JobClientset, job.Name, job.Namespace)
-		if !found {
+		_, err := c.JobClientset.BatchV1().Jobs(job.Namespace).Get(job.Name, metav1.GetOptions{})
+		if err != nil {
```
Comment (jkatz) [inline internal/controller/job/rmdatahandler.go:68]:
```suggestion
		if _, err := c.JobClientset.BatchV1().Jobs(job.Namespace).Get(job.Name, metav1.GetOptions{}); err != nil {
```

File: internal/apiserver/backrestservice/backrestimpl.go [hunk 8]
```diff
@@ -33,7 +33,7 @@ import (
 	msgs "github.com/crunchydata/postgres-operator/pkg/apiservermsgs"
 	log "github.com/sirupsen/logrus"
 	v1 "k8s.io/api/core/v1"
-	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
[the remark this answers is not in the document]
Comment (cbandy) [reply internal/apiserver/backrestservice/backrestimpl.go:36]: goimports goes the other way; it changes imports to match the references present in our code. We have a mix of both spellings, but the latter occurs more often and matches the style used by kubernetes and our generated clients.

I started by leaving the spelling alone when it was already present, but the inconsistency became irritating. This definitely could trip us up when back-patching, but any API calls after this point require scrutiny.

I also considered tucking some of this away as a veneer in `kubeapi` for a handful of reasons (e.g. delete propagation) but I can't tell yet if it's worth it.

**Pull request merged**